src/components/AreaDisplay/AreaDisplay.js

```diff
@@ -2,42 +2,217 @@
 import React, {useState, useEffect} from 'react';
 import {map, find} from 'lodash';
-import {parseRegions, filterRegions} from '../../helpers/GeographyHelpers.js';
+import {parseRegions, regionListUnion} from '../../helpers/GeographyHelpers.js';
+import _ from 'lodash';
 
 function AreaDisplay({onChangeRegion, region}) {
-  const [regions, setRegions] = useState([]);
-  const [selectedRegion, setSelectedRegion] = useState([]);
+  const [basins, setBasins] = useState([]);
+  const [selectedBasin, setSelectedBasin] = useState(null);
+  
+  const [watersheds, setWatersheds] = useState([]);
+  const [selectedWatershed, setSelectedWatershed] = useState([]);
+
+  const [conservationUnits, setConservationUnits] = useState([]);
+  const [selectedConservationUnit, setSelectedConservationUnit] = useState([]);
+  
+  const [taxons, setTaxons] = useState([]);
+  const [selectedTaxons, setSelectedTaxons] = useState([]);
+ 
+  // Hooks for this component are a little complex
+  // This component allows a user to select a "region" and the rest of the app
+  // (maps, graphs) will provide information based on what was selected. The 
+  // rest of the app is passed region information by the onChangeRegion callback
+  // function.
+  // This particular component understands three kinds of "region" - "basins", 
+  // "conservation units", and "watersheds". Basins represent large river systems
+  // like the Nass or Columbia; watersheds represent a stream network as defined by the 
+  // BC Freshwater Atlas, and Conservation Units represent the location where a
+  // particular salmon species lives. A user may examine data for any of these region kinds.
+  // While other components don't know or care what kind of "region" a user has selected,
+  // this component has some extra hooks relating to the hierarchical nature of
+  // region selection. A "basin" contains many "watersheds" and "conservation units"
+  // and when a user selects a basin, in addition to that basin being made the selected 
+  // region and broadcast to the rest of the app, the conservation unit and watershed
+  // selector are updated to include only regions in that basin. This is to help a user 
+  // find the conservation unit or watershed they care about - they may not know its
+  // official name, but they probably know what river system it's part of.
+  //
+  // Users may also indicate which salmon species or taxons they ae interested in.
+  // This doesn't change the selected basin, conservation unit, or watershed, but it
+  // does change which ones are selectable in the dropdown menus - only regions that 
+  // contain the currently selected species are available.
 
-  //fetch region list from the API.
+  // fetch basin list from the API.
   // this only needs to be done once, when the component is loaded
   useEffect(() => {
-    if(regions.length === 0) {
-        getWatersheds().then(
+      if(basins.length === 0) {
+          getBasins().then(
```

Comment:
Possible bug here: If the backend returns a zero-length list, then you enter an infinite loop of requests. Unlikely but not strictly speaking impossible.